### PR TITLE
Upgrade Ubuntu version from `18.04` to `20.04`

### LINKED ISSUE #11363: Use Ubuntu LTS 20.04 instead of 18.04 in CI
## 🚀 Enhancement in CI
Some are already been running on 20.04, but not all of them (see below).

### Motivation
It enables testing on a more widely used version of Ubuntu.

### Pitch
Replace the following occurrences with 20.04:
```console
$ grep -rnI 18.04
./dockers/base-cuda/Dockerfile:17:FROM nvidia/cuda:${CUDA_VERSION}-devel-ubuntu18.04
./dockers/base-conda/Dockerfile:17:FROM nvidia/cuda:${CUDA_VERSION}-devel-ubuntu18.04
./.github/workflows/ci_test-full.yml:20:        os: [ubuntu-18.04, windows-2019, macOS-10.15]
./.azure-pipelines/gpu-tests.yml:29:      # base ML image: mcr.microsoft.com/azureml/openmpi3.1.2-cuda10.2-cudnn8-ubuntu18.04
```

### Alternatives
No update

### Additional context
Ubuntu's release cycle: https://ubuntu.com/about/release-cycle

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes #11363.

~FYI, currently, `ubuntu-latest` is an alias to `ubuntu-20.04` as documented in https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources.~

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Comment by akihironitta** [comment]
`nvidia/cuda:cuda10.2-devel-ubuntu20.04` image doesn't exist, so I reverted the change to the dockerfiles in 51560db. 
https://hub.docker.com/r/nvidia/cuda/tags?page=1&name=10.2-devel-ubuntu

File: .github/workflows/ci_test-full.yml [hunk 1]
```diff
@@ -17,13 +17,13 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-18.04, windows-2019, macOS-10.15]
+        os: [ubuntu-latest, windows-2019, macOS-10.15]
```
Comment (Borda) [inline .github/workflows/ci_test-full.yml:20]: I would be rather explicit or at least add a link to GH docs where is the mapping - at latest really is
there could be some confusion as actually, latest is 21 but for gh actions, it is 20
```suggestion
        os: [ubuntu-20.04, windows-2019, macOS-10.15]
```

Comment (akihironitta) [reply .github/workflows/ci_test-full.yml]: Totally makes sense :]

**Review by Borda — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]
